lib/ecl/ecl_kw.c

```diff
@@ -866,11 +868,18 @@ void ecl_kw_iset_string_ptr( ecl_kw_type * ecl_kw, int index, const char * s) {
   if(input_len > type_len)
     util_abort("%s: String of length %d cannot hold input string of length %d\n", __func__, type_len, input_len);
 
-  char * ecl_string = (char *) ecl_kw_iget_ptr(ecl_kw, index);
-  for(int i = 0; i < input_len; ++i)
-    ecl_string[i] = s[i];
+  {
+    char * ecl_string = (char *) ecl_kw_iget_ptr(ecl_kw, index);
+    int i;
+
+    for(i = 0; i < input_len; ++i)
+      ecl_string[i] = s[i];
 
-  ecl_string[input_len] = '\0';
+    for (i=input_len; i < type_len; ++i)
+      ecl_string[i] = ' ';
+
+    ecl_string[type_len] = '\0';
+  }
```

Comment:
I think the following is more readable:

```c
for (int i = 0; i < type_len; ++i) {
  ecl_string[i] = (i < input_len) ? s[i] : ' ';
}
```
And also, you don't need the `{` and the `}` guards here.